core/shared/src/main/scala/org/http4s/Message.scala

```diff
@@ -298,6 +300,12 @@ final class Request[F[_]] private (
       attributes = attributes,
     )
 
+  def normalize: Request[F] =
```

Comment:
:thinking: A couple thoughts here.

Bodies are not forbidden with GET requests, but have undefined semantics and servers _must_ ignore them. That said, I wonder if the spec permits an encoding we should also permit it as well, but just heavily discourage the use. 

In any case, should we yield an error if the components of the message are invalid, rather than silently clean them up?

Reply:
The original bug (#6108) says that while HTTP allows this, some backends (e.g., okhttp-client) don't.  The test guarantees that the same request is processed the same way by a server, regardless of how the client chooses to encode it.

Since a complaint HTTP server MUST ignore the body, a test against a compliant HTTP server can't further specify that a body is or is not sent.  We could check the entity with a raw TCP server, but since the distinction isn't observable through HTTP, should we?  Normalizing everything makes things more consistent, but I think maybe all we need to do is get all the backends passing this new test.  Some backends will have to normalize, and others, :shrug:.

Reply:
Yeah, the client backends seem to be too different for a single central normalization to work nicely. E.g. while reading a bit more, I came across this example https://www.elastic.co/guide/en/elasticsearch/reference/8.1/search-search.html#search-search-api-example - so maybe there are http4s users out there who have selected their client to allow this kind of request and those clients shouldn't break.